src/trackers/classes/request.js

```diff
@@ -16,6 +18,18 @@ class Request {
         this.originalSubdomain = undefined
         this.responseHash = reqData.responseBodyHash
         this.nameservers = []
+        this.firstPartyCookies = site.documentCookies
+            .filter(cookie => cookie.source === reqData.url && (!cookie.domain || cookie.domain.endsWith(site.domain)) && cookie.value && cookie.value.length > COOKIE_LENGTH_CUTOFF)
```

Comment:
hm, this is a bit hard to read so I'd break it down if possible. Two thoughts:

> cookie.domain.endsWith(site.domain)

If cookie domain is "hub.com" and site is "github.com" those would incorrectly match, right?

> cookie.value.length > COOKIE_LENGTH_CUTOFF

Isn't filtering it out here messing up the `cookie.uniqueness = cookie.values.size / cookie.pages` stat?

Reply:
> If cookie domain is "hub.com" and site is "github.com" those would incorrectly match, right?

Hmm... this condition kind of got built up as I worked on it, and is now a bit confusing. I'll split into a separate function with tests to make sure it works correctly in cases like you outlined.

> > cookie.value.length > COOKIE_LENGTH_CUTOFF
> 
> Isn't filtering it out here messing up the `cookie.uniqueness = cookie.values.size / cookie.pages` stat?

I assume that it is unlikely a script would used both long and short values for an identifier (which is what this analysis is concerned with). The filtering is intended to cut out less interesting cookies early on, to reduce the about of data to process, and the number of irrelevant cookies in the final results. I would also be ok with removing it though and seeing what this looks like.

Reply:
I updated this filter condition to use a function for the 1st party check and added tests for it.

Reply:
> I assume that it is unlikely a script would used both long and short values for an identifier

If it's not a big deal to check this I think we should.

Reply:
Yep, I already removed this (https://github.com/duckduckgo/tracker-radar-detector/pull/28/commits/913c7c1a9ed76980bc2ae96b173a1e7303b09a03) and now doing a test run to ensure that this doesn't cause any issues